core/src/main/java/org/hibernate/ogm/datastore/impl/AvailableDatastoreProvider.java

```diff
@@ -19,6 +19,7 @@
 	EHCACHE( "org.hibernate.ogm.datastore.ehcache.impl.EhcacheDatastoreProvider" ),
 	MONGODB( "org.hibernate.ogm.datastore.mongodb.impl.MongoDBDatastoreProvider" ),
 	FONGO( "org.hibernate.ogm.datastore.mongodb.impl.FongoDBDatastoreProvider" ),
+	NEO4J( "org.hibernate.ogm.datastore.neo4j.remote.impl.RemoteNeo4jDatastoreProvider" ),
```

Comment:
Yeah well, my problem is that it's not consistent at all with the code where Neo4j* is for embedded (most of the time) and RemoteNeo4j* for remote.

I'll post a general comment on this at the end of my review so that we can deal with this.